pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -440,12 +453,81 @@ func (m *cgroupManagerImpl) Update(cgroupConfig *CgroupConfig) error {
 		libcontainerCgroupConfig.PidsLimit = *cgroupConfig.ResourceParameters.PidsLimit
 	}
 
+	if m.adapter.cgroupManagerType == libcontainerSystemd {
+		if err := m.updateSlicePropertiesSystemd(libcontainerCgroupConfig); err != nil {
+			return fmt.Errorf("failed to slice properties for unit %v: %v", cgroupConfig.Name, err)
+		}
+	}
 	if err := setSupportedSubsystems(libcontainerCgroupConfig); err != nil {
 		return fmt.Errorf("failed to set supported cgroup subsystems for cgroup %v: %v", cgroupConfig.Name, err)
 	}
 	return nil
 }
 
+// Since libcontainer doesn't support updating slice properties, and only sets resources params during
+// creating of slice, we have to update them manually.
+//
+// The code is adapted from:
+// https://github.com/opencontainers/runc/blob/v1.0.0-rc10/libcontainer/cgroups/systemd/apply_systemd.go#L138
+func (m *cgroupManagerImpl) updateSlicePropertiesSystemd(c *libcontainerconfigs.Cgroup) error {
+	var (
+		unitName   = c.Name
+		properties []systemddbus.Property
+	)
+
+	if c.Resources.Memory != 0 {
+		properties = append(properties,
+			newProp("MemoryLimit", uint64(c.Resources.Memory)))
+	}
+
+	if c.Resources.CpuShares != 0 {
+		properties = append(properties,
+			newProp("CPUShares", c.Resources.CpuShares))
+	}
+
+	// cpu.cfs_quota_us and cpu.cfs_period_us are controlled by systemd.
+	if c.Resources.CpuQuota != 0 && c.Resources.CpuPeriod != 0 {
+		// corresponds to USEC_INFINITY in systemd
+		// if USEC_INFINITY is provided, CPUQuota is left unbound by systemd
+		// always setting a property value ensures we can apply a quota and remove it later
+		cpuQuotaPerSecUSec := uint64(math.MaxUint64)
+		if c.Resources.CpuQuota > 0 {
+			// systemd converts CPUQuotaPerSecUSec (microseconds per CPU second) to CPUQuota
+			// (integer percentage of CPU) internally.  This means that if a fractional percent of
+			// CPU is indicated by Resources.CpuQuota, we need to round up to the nearest
+			// 10ms (1% of a second) such that child cgroups can set the cpu.cfs_quota_us they expect.
+			cpuQuotaPerSecUSec = uint64(c.Resources.CpuQuota*1000000) / c.Resources.CpuPeriod
+			if cpuQuotaPerSecUSec%10000 != 0 {
+				cpuQuotaPerSecUSec = ((cpuQuotaPerSecUSec / 10000) + 1) * 10000
+			}
+		}
+		properties = append(properties,
+			newProp("CPUQuotaPerSecUSec", cpuQuotaPerSecUSec))
+	}
+
+	if c.Resources.BlkioWeight != 0 {
```

Comment:
is it currently used by the kubelet or can we drop it?  For cgroup v2 we would need to convert to a different range and use the `IOWeight=` property